companion/flashing.md

```diff
@@ -41,43 +63,77 @@ Zadig will install the driver, and should report success. When done you can dism
-button and select a location for the file. This will automatically  
-build and download the latest available version of OpenTX for Taranis.  
-Close the Preferences window, and select the Burn -&gt; Write Flash  
-memory command. Make sure your radio is powered off and connected.  
-Browse for the file you just downloaded. It will be recognised  
-automatically and the version will be displayed.  If you wish to  
-replace the radio's splash screen, you can either load one from a file  
-or select it from the available library, and click the "Substitute  
-image in firmware" box.  Click "Burn to TX".  Once the progress bar is  
-done and the success prompt is shown you can dismiss it and disconnect  
-your radio :\) If you get an error about the dfu-util executable not  
-being found, open the flashing tool settings with the Burn -&gt;  
-Configure menu item. Browse for the dfu-util executable, which should  
-be: On Windows: In the OpenTX companion installation folder \(by  
-default C:\Program Files\companion9x\dfu-util.exe on 32-bit systems,  
-and C:\Program Files \(x86\)\companion9x\dfu-util.exe on 64-bit  
-systems\).  On Mac OS: `/Applications/OpenTX Companion  
-2.2.app/Contents/Resources/dfu-util`
+## Downloading The OpenTX Firmware
+
+The only recommended way to download the OpenTX firmware is using Companion. This process compiles/builds the firmware on the OpenTX servers and downloads a copy to the computer.
+
+To download the firmware follow these steps:
+
+* Radio Profile: Make sure the correct transmitter profile is selected
+* Build Options: Make sure the build options selected are the ones you want
+* Download Firmware: File -> Download
+* Click on Download firmware (The download could take a few minutes)
+* Save the firmware .bin file on the computer
+* You may be prompted to write the firmware to the transmitter
+* If you want to choose yes connect the transmitter to the computer in bootloader mode
+* Click on Yes
+* Select Check Hardware compatibility to make sure the firmware being written to the transmitter is for that particular hardware.
+* Click on Write to TX
+* Eject the mounted drives
+* Disconnect the USB cable
+* Power the transmitter off
+
+# Writing The OpenTX Firmware
+
+Connect the transmitter to the computer in bootloader mode.
+
+There are two options to write the firmware:
+* Using The Bootloader Menu
+* Using Companion
+
+## Using The Bootloader Menu
+
+* Place a copy of the firmware on the sd card in the FIRMWARE folder
+* Rename the firmware file so that the filename including the .bin extension is 35 characters or less.
+* Eject the two mounted drives
+* Disconnect the USB cable
+* Choose Write Firmware from the bootloader menu
+* Select the firmware to be written
+* Press ENT
+* Hold ENT down to start writing the firmware
+* When the “Writing Complete” message is displayed press EXIT
+* Power the transmitter off
+
+## Using Companion (dfu-util)
```

Comment:
Ahh... I went back up and noticed this table which is not how I wrote it....
+| Tx  | OpenTX Bootloader | Flash method |
 +| --- | ------------------- | ------  | 
 +| X7 | 2.2 |  Bootloader, Dfu-util |
 +| X9D | 1.x | Dfu-util |
 +| X9D | 2.x |  Bootloader, Dfu-util |
 +| X9D+ | 2.x | Any |  Bootloader, Dfu-util |
 +| X9E | 2.x | Any |  Bootloader, Dfu-util |
 +| X12S | 2.2 + later |  Dfu-util |

I wrote is as
**Is Zadig required**

![bootloadersmall](https://user-images.githubusercontent.com/20209851/29190168-1b415e06-7de7-11e7-8d92-0010df929bbd.png)


| Tx  | OpenTX Boot Loader | PC OS | Add PC Driver |
| --- | ------------------- | ------  | ----------------------- |
| X7 | 2.2 | Any | No |
| X9D | 1.x | Windows | Zadig |
| X9D | 1.x | Linux | Dfu-util |
| X9D | 1.x | Mac | Dfu-util |
| X9D | 2.x | Any | No |
| X9D+ | 2.x | Any | No |
| X9E | 2.x | Any | No |
| X12S | 2.2 + later | Windows | Zadig |
| X12S | 2.2 + later| Linux | Dfu-util |
| X12S | 2.2 + later| Mac | Dfu-util |

The difference is
- do I as a user need to install a driver

versus
- telling me the user what is being used....

I wrote the rest of the instructions in the context on how to get something done without the what is being used.... 

So should I just close this pull request?
Or will someone fix its...
The way it is now we have mixed two different approaches which is beyond me to fix...

Reply:
Your last table looks good. 

Still I hope you realize that for example a 9XE with 2.x can also be written with dfu-util. Perhaps your way of mentioning only **no driver needed** is more appropriate since it is less confusing to the user. But the fact is that all radios in the above table regardless of the OpenTX version installed support dfu-util/zadig method, just that some of them also support  **no driver needed** method.